src/classes/RD2_OpportunityEvaluation_TEST.cls

```diff
@@ -328,6 +328,37 @@ public with sharing class RD2_OpportunityEvaluation_TEST {
         testNextInstallmentOppCreation(rd);
     }
 
+    @isTest
+    private static void shouldNotRunTheBatchIfAnotherJobIsRunning() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+        RD2_Settings_TEST.configureForASyncFirstOpportunityCreate();
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder()
+            .withStatusLapsed()
+            .build();
+
+        final Date prevCloseDate = START_DATE.addMonths(1);
+        final Date today = prevCloseDate.addDays(1);
+        final Date nextCloseDate = START_DATE.addMonths(2);
+        RD2_ScheduleService.currentDate = today;
+
+        insert rd;
+
+        insert getOpportunityBuilder(rd)
+            .withOpenStage()
+            .withCloseDate(prevCloseDate)
+            .build();
+
+        List<Opportunity> opps = oppGateway.getRecords(rd);
+        System.assertEquals(1, opps.size(), 'An Opp should exist: ' + opps);
+
+        RD2_OpportunityEvaluation_BATCH batch  = new RD2_OpportunityEvaluation_BATCH(nextCloseDate.addDays(1));
+        batch.anotherBatchIsInProgress = true;
+        runBatch(batch);
+
+        System.assertEquals(1, [SELECT id FROM Error__c].size(), 'Error should have been created');
```

Comment:
The `errorGateway` can be used to get error instead of writing SOQL directly. The expected error message should be verified. The similar is applicable to the new legacy batch test in `RD_RecurringDonations_TEST` .

```
        List<Error__c> errors = errorGateway.getRecords();
        System.assertEquals(1, errors.size(), 'One error should be created: ' + errors);
        
        String expectedMessage = String.format(
            System.Label.stgConcurrentBatchError,
            new String[]{ RD2_OpportunityEvaluation_BATCH.class.getName() }
        );
        System.assertEquals(expectedMessage, errors[0].Full_Message__c, 'Error message should match');
```